cache/reference_counted.go

```diff
@@ -35,17 +34,25 @@ type ParamsWithDisposal struct {
 }
 
 // shardedCacheWithDisposal shards a CacheWithDisposal across its keys.  It requires that its
-// keyus have a String() method that is compatible with their equality, i.e. k1.GoString() ==
+// keys have a String() method that is compatible with their equality, i.e. k1.GoString() ==
 // k2.GoString() implies k1 equals k2 in the sense of map comparison.
 type shardedCacheWithDisposal struct {
 	name   string
 	Seed   maphash.Seed
 	Shards []CacheWithDisposal
 }
 
+const (
+	DefaultCacheWithDisposalShards = 1
+	DefaultCacheWithDisposalSize   = 1000
+)
+
 func NewCacheWithDisposal(p ParamsWithDisposal) *shardedCacheWithDisposal {
 	if p.Shards <= 0 {
-		panic("need at least 1 shard")
+		p.Shards = DefaultCacheWithDisposalShards
+	}
```

Comment:
This is almost always wrong.  Instead, we create from configuration and force a minimum *there*.  (I made that change concurrently with your branch, so it wasn't there initially).
E.g. 0 might be better used as a magic value that says "don't cache".

Reply:
removed the default size and added no cache option